registry/handlers/api_test.go

```diff
@@ -815,6 +818,59 @@ func TestManifestAPI(t *testing.T) {
 	testManifestAPIManifestList(t, env2, schema2Args)
 }
 
+func TestGetManifestWithStorageError(t *testing.T) {
+	const repositoryWithManifestNotFound = "manifesttagnotfound"
+	const repositoryWithManifestInvalidPath = "manifestinvalidpath"
+	const repositoryWithManifestBadLink = "manifestbadlink"
+	const repositoryWithGenericStorageError = "genericstorageerr"
+
+	manifestPathRepositoryWithManifestNotFound := fmt.Sprintf("%s/_manifests/tags", repositoryWithManifestNotFound)
+	manifestPathRepositoryWithManifestInvalidPath := fmt.Sprintf("%s/_manifests/tags", repositoryWithManifestInvalidPath)
+	manifestPathRepositoryWithManifestBadLink := fmt.Sprintf("%s/_manifests/tags", repositoryWithManifestBadLink)
+	errGenericStorage := errors.New("generic storage error")
+
+	deleteEnabled := false
+	env1 := newTestEnvWithMockStorageDriver(t, deleteEnabled)
+	defer env1.Shutdown()
+	testDriver := env1.app.driver.(*testdriver.MockDriver)
+	// Override the GetContent to generate errors
+	testDriver.MockGetContent = func(ctx context.Context, path string) ([]byte, error) {
```

Comment:
Declare the type here and assign an error per path. No need to have a separate package, function pointers or any other nonsense.

Literally, just declare the type here and make it have this implementation.

I know you think "Someone will use this later", but they won't and we'll end up with yet another mock. Make mocks narrow.

```go
type mockErrorDriver struct {
  returns []struct {
    partial
    content []byte
    err error
  }
}

func (dr *mockErrorDriver) GetContent(ctx context.Context, path string) ([]byte, error) {
  for _, returns := range dr.returns {
    if strings.Contains(path, returns.partial) {
      return returns.content, returns.err
    }
  }
 return nil, errGenericStorage
}
```